MMOCoreORB/src/server/zone/managers/player/creation/PlayerCreationManager.cpp

```diff
@@ -328,11 +328,30 @@ bool PlayerCreationManager::createCharacter(ClientCreateCharacterCallback* callb
 
 	auto client = callback->getClient();
 
-	if (client->getCharacterCount(zoneServer.get()->getGalaxyID()) >= 2) {
-		ErrorMessage* errMsg = new ErrorMessage("Create Error", "You are limited to 2 characters per galaxy.", 0x0);
-		client->sendMessage(errMsg);
+	try {
+		StringBuffer query1;
+		uint32 accountId1 = client->getAccountID();
+		query1 << "SELECT admin_level as t FROM accounts as a WHERE a.account_id = " << accountId1 << ";";
 
-		return false;
+		UniqueReference<ResultSet*> res1(ServerDatabase::instance()->executeQuery(query1));
+		uint32 adm_lvl = 0;
+		if (res1 != nullptr && res1->next()) 
+			adm_lvl = res1->getUnsignedInt(0);
+
+		if (adm_lvl >= 2 && client->getCharacterCount(zoneServer.get()->getGalaxyID()) >= 3) {
```

Comment:
I'm not so sure using the `admin_level` is a great idea to manage this permission. While yes, only level [9, 10, 12, and 15 will update the created characters' permissions](https://github.com/swgintended/Core3/blob/eaeacdb2464aceead5e0fb376e07edc7a1ef440e/MMOCoreORB/src/server/zone/managers/player/creation/PlayerCreationManager.cpp#L468-L470), that could change at some point in the future and would result in all characters of the account being QA-level admins.

Security and code-wise, it just seems like a bad idea.

What seems smarter would be to add a `character_limit` column to the account table and the [Account.idl](https://github.com/swgintended/Core3/blob/unstable/MMOCoreORB/src/server/login/account/Account.idl) class to manage this.

```cpp
int limit = AccountManager::instance()->getAccount(client->getAccountID())->getCharacterLimit();
if (limit != -1 && client->getCharacterCount(zoneServer.get()->getGalaxyID()) >= limit) {
  ErrorMessage* errMsg = new ErrorMessage("Create Error", "You are limited to " + String::valueOf(limit) + " characters per galaxy.", 0x0);
  client->sendMessage(errMsg);
  return false;
}
```

You can add this to the end of swgemu.sql and run it locally to modify the table.
```sql
--
-- SWGIntended
-- Add character limit to accounts
--
ALTER TABLE `swgemu`.`accounts` ADD COLUMN `character_limit` tinyint(1) NOT NULL DEFAULT 2 AFTER `salt`;
```